packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -0,0 +1,868 @@
+    createSplitArgs: CreateSplitV2Config,
+  ): Promise<{
+    splitAddress: Address
+  }> {
+    if (!createSplitArgs.controller) createSplitArgs.controller = zeroAddress
+    if (!createSplitArgs.creator) createSplitArgs.creator = zeroAddress
+
+    const { addresses, allocations } = getAddressAndAllocationFromRecipients(
+      createSplitArgs.recipients,
+    )
+    const totalAllocation = PERCENTAGE_SCALE
+
+    validateAddress(createSplitArgs.controller)
+    validateAddress(createSplitArgs.creator)
+    addresses.map((recipient) => validateAddress(recipient))
+
+    this._requirePublicClient()
+
+    const factory = this._getSplitV2FactoryContract(
+      createSplitArgs.splitType ?? SplitV2Type.Pull,
+    )
+
+    let splitAddress
+    if (createSplitArgs.salt) {
+      splitAddress = await factory.read.predictDeterministicAddress([
+        {
+          recipients: addresses,
+          allocations,
+          totalAllocation: totalAllocation,
+          distributionIncentive: getNumberFromPercent(
+            createSplitArgs.distributorFeePercent,
+          ),
+        },
+        createSplitArgs.controller,
+        createSplitArgs.salt,
+      ])
+    } else {
+      splitAddress = await factory.read.predictDeterministicAddress([
+        {
+          recipients: addresses,
+          allocations,
+          totalAllocation: totalAllocation,
+          distributionIncentive: getNumberFromPercent(
+            createSplitArgs.distributorFeePercent,
+          ),
+        },
+        createSplitArgs.controller,
+      ])
+    }
+
+    return {
+      splitAddress,
+    }
+  }
+
+  async isDeployed(createSplitArgs: CreateSplitV2Config): Promise<{
+    splitAddress: Address
+    deployed: boolean
+  }> {
+    if (!createSplitArgs.controller) createSplitArgs.controller = zeroAddress
+    if (!createSplitArgs.creator) createSplitArgs.creator = zeroAddress
+
+    const { addresses, allocations } = getAddressAndAllocationFromRecipients(
+      createSplitArgs.recipients,
+    )
+
+    validateAddress(createSplitArgs.controller)
+    addresses.map((recipient) => validateAddress(recipient))
+    this._requirePublicClient()
+
+    const factory = this._getSplitV2FactoryContract(
+      createSplitArgs.splitType ?? SplitV2Type.Pull,
+    )
+
+    const totalAllocation = PERCENTAGE_SCALE
+
+    if (!createSplitArgs.salt) throw new Error('Salt required')
+    const [splitAddress, deployed] = await factory.read.isDeployed([
+      {
+        recipients: addresses,
+        allocations,
+        totalAllocation: totalAllocation,
+        distributionIncentive: getNumberFromPercent(
+          createSplitArgs.distributorFeePercent,
+        ),
+      },
+      createSplitArgs.controller,
+      createSplitArgs.salt,
+    ])
+
+    return {
+      splitAddress,
+      deployed,
+    }
+  }
+
+  async getSplitBalance(
+    splitAddress: Address,
+    token: Address,
+  ): Promise<{
+    splitBalance: bigint
+    warehouseBalance: bigint
```

Comment:
nit: just wanted to mention this idea. is it worth throwing `splitMainBalance` in here too?

My gut says no, we can make some separate functions that make it easy to check balances in `splitMain` and `splitWarehouse` and do a withdrawal pre distribution (if necessary). Don't think this function is right for that. But just wanted to mention it since it crossed my mind.

Reply:
yeah, I'd say lets keep it separate